### DISCUSSION TITLE
Decrypted and Encrypted SSL traffic on bond0 monitoring interface

### DISCUSSION THREAD

**Discussion by sedoy107** [Security-Onion-Solutions/securityonion#7637] [Unsupported Versions] [answered]
Good day folks. Have just setup SO 2 and plugged three of my sniffing interfaces into the box where one of them delivers a decrypted SSL traffic. After a few minutes of looking at the alerts, I realized that Suricata has not been reporting HTTP over HTTPS port rule which was the case on SO 16.

I looked at the way the monitoring interface is made and realized that all of the ensxxx NICs are bridged into bond0 which is then monitored by Suricata. I did a quick tcpdump on bond0 and I was able to see the decrypted traffic. Then, I tried to unplug the interface with the original, encrypted, TLS traffic and I started seeing alerts for decrypted traffic.

As far as I understand, and that is a rough guess, Suricata inspects the original traffic first as it is live traffic, then, with a slight delay, the decrypted traffic arrives at its dedicated sniffing interface from the decryptor. Given that the decrypted and encrypted traffic is the same thing in terms of the 5-tuple, I'd dare to guess that when the decrypted traffic makes it to the Suricata engine the session for that traffic has already been processed and let go of.

I wonder if there is a tweak to trick Suricata into processing the decrypted tls traffic while it is on the same monitoring interface along with its encrypted counterpart. Has anyone come across a similar issue?

PS
That setup breaks the NSM storage optimization where one can disable recording TLS traffic into the PCAP via the BPF. Having a dedicated sensor for decrypted might be a better solution. That's what I run in my lab for now

**Comment by dougburks** [comment, marked as answer]
I'm not sure I understand your configuration. Seems like if you have the capability to send decrypted SSL traffic to Security Onion, then you wouldn't want to send the encrypted version of it at all.

**Reply by sedoy107** [reply]
The reason I have both, encrypted and decrypted is that I'd like to keep the TLS metadata. Say, I want to analyze the the cert presence on the network and find outliers or rogue certs. 

Also, not every HTTPS is being decrypted and for those connections would still be good to have metadata for TLS connections.

**Reply by dougburks** [reply]
If that's the case, then I would agree with what you said above about having a dedicated sensor for decrypted traffic.

**Reply by sedoy107** [reply]
Ok, take that as an answer. Thank you Doug!
